Content/ReleaseBriefings/4_6/Section_NewFeatures.tex

```diff
@@ -0,0 +1,156 @@
+%            std::cout << "hello world\n";
+%        }
+%    \end{code}
+%\end{frame}
+
+%\begin{frame}[fragile]{Example table}
+%    \begin{center}
+%        \begin{tabular}{l|l}
+%            a & b \\\hline
+%            c & d
+%        \end{tabular}
+%    \end{center}
+%\end{frame}
+
+%==========================================================================
+
+% \begin{frame}[fragile]\label{sec:new_features}
+
+  % {\Huge Kokkos::Graph features}
+
+  % \vspace{10pt}
+
+% \end{frame}
+
+\begin{frame}[fragile]{Kokkos::Graph recap}
+ \begin{itemize}
+     \item describes asynchronous workloads organised as a direct acyclic graph (DAG)
+     \item executed using \texttt{submit()}, possibly many times, observing dependencies
+      \begin{code}[keywords={auto}]
+auto graph = Kokkos::create_graph([&](auto root) {
+    auto node_A = root.then_parallel_for("A", ...policy..., ...functor...);
+    auto node_B = node_A.then_parallel_for("B", ...policy..., ...functor...);
+    auto node_C = node_A.then_parallel_for("C", ...policy..., ...functor...);
+
+    auto node_D = Kokkos::when_all(node_B, node_C).
+                  then_parallel_for("D", ...policy..., ...functor...);
+});
+
+graph.instantiate();
+
+graph.submit();
+      \end{code}
+ \end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]{Kokkos::Graph new features}
+ \begin{itemize}
+  \item \href{https://github.com/kokkos/kokkos/pull/7629}{\texttt{then} node}: executes a callable on device
+   \item Executed in the \texttt{ExecutionSpace} the graph is submitted to
+     \begin{code}[keywords={auto}]
+auto graph = Kokkos::create_graph([&](auto root) {
+    auto node_A = root.then_parallel_for("A", ...policy..., ...functor...);
+    auto node_B = node_A.then(...functor...);
+});
+
+graph.instantiate();
+
+graph.submit();
+     \end{code}
+ \end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]{Kokkos::Graph new features}
+ \begin{itemize}
+   \item \href{https://github.com/kokkos/kokkos/pull/7664}{interoperability}: create a \texttt{Kokkos::Graph} from a native Cuda/HIP/Sycl graph
+   \item Does not yet allow linking Kokkos nodes to native nodes
+     \begin{code}[keywords={auto}]
+cudaGraph_t native_graph = nullptr;
+KOKKOS_IMPL_CUDA_SAFE_CALL(cudaGraphCreate(&native_graph, 0));
+auto graph_from_native =
+  Kokkos::Experimental::create_graph_from_native(exec, native_graph);
+auto root = Kokkos::Impl::GraphAccess::create_root_ref(graph_from_native);
+root.then_parallel_for(1, Increment<view_t>{data});
+...
+graph_from_native.submit(exec);
+     \end{code}
+ \end{itemize}
+\end{frame}
+
+% \begin{frame}[fragile]\label{sec:new_features}
+
+  % {\Huge Multi-GPU for HIP Backend}
+
+  % \vspace{10pt}
+
+% \end{frame}
+
+\begin{frame}[fragile]{Multi-GPU for HIP Backend}
+  \begin{itemize}
+    \item Launch kernels on multiple devices from a single host process
+    \item Available for ROCm 5.6 and later
+    \item Requires knowledge of HIP API functions for creating and destroying streams
```

Comment:
knowledge?
change to "direct use of HIP runtime API"?

Reply:
You might want to mention setting the device on which the stream will be created and manage the stream lifetime

Reply:
Updated. I put setting device and creating/destroying streams in the example. I may just mention verbally then.